.pre-commit-config.yaml

```diff
@@ -56,20 +56,15 @@ repos:
         hooks:
               - id: pydocstyle
                 args: ["--config=python/.flake8"]
-      - repo: local
+      - repo: https://github.com/pre-commit/mirrors-clang-format
+        rev: v11.1.0
         hooks:
               - id: clang-format
-                # Using the pre-commit stage to simplify invocation of all
-                # other hooks simultaneously (via any other hook stage).  This
-                # can be removed if we also move to running clang-format
-                # entirely through pre-commit.
-                stages: [commit]
-                name: clang-format
-                description: Format files with ClangFormat.
-                entry: clang-format -i
-                language: system
                 files: \.(cu|cuh|h|hpp|cpp|inl)$
-                args: ['-fallback-style=none']
+                types_or: [file]
+                args: ['-fallback-style=none', '-style=file', '-i']
```

Comment:
Note that without `-style=file`, the clang-format mirror will not handle `.inl` file type.

Reply:
Actually, I've added `inl` into `c++` category (https://github.com/pre-commit/identify/pull/273) so this can be simpler:
```yaml
types_or: [c++, cuda]
args: ['-fallback-style=none', '-i']
```
where we don't need `files: \.(cu|cuh|h|hpp|cpp|inl)$` anymore.

Reply:
Great! I can make that change.

Reply:
Ah, right. We'll leave it as-is.

Reply:
Sounds good to me.

Reply:
So I think the original solution is the best path forward until we require a sufficiently new version of pre-commit, correct?

Reply:
Yes. We will specify `files: \.(cu|cuh|h|hpp|cpp|inl)$`.

The "sufficiently new" requirement actually applies to `identify`, not `pre-commit`. The current version of `pre-commit` only requires `identify>=1.0.0` so there's no definitive timeline for when we can guarantee support for inl files based only on `pre-commit` versions. We would need a second pinning for `identify`. The status quo is sufficient for now.

Reply:
Oh oops my page hadn't refreshed with the two comments prior to my last one. Yes, we would need to pin identify. I don't think that's a problem though. It's fine to pin dependencies of dependencies if there's a good reason to do so. We can decide on that at some later point, though.